src/js/ble_uuid_util.js

```diff
@@ -36,8 +36,36 @@
 
 module.exports.removeDashes = function(uuid) {
   if (uuid) {
-    uuid = uuid.replace(/-/g, '');
+    var start = 0;
+    var end = 0;
+    var str = '';
+    uuid = String(uuid);
+
+    while (end < uuid.length) {
+      if (uuid.charAt(end) !== '-') end++;
+      else {
+        str += uuid.slice(start, end);
+        start = end = end + 1;
+      }
+    }
+
+    str += uuid.slice(start, end);
   }
 
-  return uuid;
+  return str;
 };
+
+module.exports.refineUuid = function(str, joint) {
```

Comment:
Originally, Tokenize was a function that slices a string into two chars as you suggested. At the advice of @zherczeg, I combined Tokenize with reverse().join(sth). In so doing it got to need more comprehensive name not just representing matching regular expressions but also including its other operations. In this regard, Could you suggest an alternative again? I admit that refine is too ambiguous and not general for uuid as you think.